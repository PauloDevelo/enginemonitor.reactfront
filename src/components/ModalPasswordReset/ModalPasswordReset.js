import React, { useState } from "react";
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { faUnlockAlt } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { CSSTransition } from 'react-transition-group';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';

import changePasswordMsg from "./PasswordReset.messages";

import MyForm from "../Form/MyForm"
import MyInput from "../Form/MyInput"
import Alerts from "../Alerts/Alerts"

import EquipmentMonitorService from '../../services/EquipmentMonitorServiceProxy';
import HttpError from '../../http/HttpError'

import '../../style/transition.css';

const ModalPasswordReset = ({visible, className, toggle, data}) => {
	const [infoMsg, setInfoMsg] = useState(undefined);
	const [resetPasswordErrors, setResetPasswordErrors] = useState(undefined);
	const [isLoading, setIsLoading] = useState(false);
    const [isError, setIsError] = useState(false);

    const cancel = () => {
		setResetPasswordErrors(undefined);
		setInfoMsg(undefined);
		toggle();
    }

    const handleSubmit = async(data) => {
		setIsError(false);
		setIsLoading(true);
		setInfoMsg(undefined);
	  
        try{
			if(data.newPassword1 === data.newPassword2){
				await EquipmentMonitorService.resetPassword(data.email, data.newPassword1);
				setResetPasswordErrors(undefined);
				setInfoMsg("confirmPasswordChange");
			}
			else{
				setIsError(true);
				setResetPasswordErrors("passwordsHaveToBeIdentical");
			}
            
		}
		catch(errors){
			setIsError(true);

			if(errors instanceof HttpError){
                const newResetPasswordErrors = errors.data;
                setResetPasswordErrors(newResetPasswordErrors);
			}
		}
		setIsLoading(false);
    }

	return (
		<CSSTransition in={visible} timeout={300} classNames="modal">
			<Modal isOpen={visible} toggle={cancel} className={className} fade={false}>
				<ModalHeader toggle={cancel}><FontAwesomeIcon icon={faUnlockAlt} />{' '}<FormattedMessage {...changePasswordMsg.modalResetPasswordTitle} /></ModalHeader>
				<ModalBody>
				    {visible && <MyForm submit={handleSubmit} id="formChangePassword" initialData={data}>
                        <MyInput name="email" 			label={changePasswordMsg.email} 			type="email"     	required="true" readonly="true" />
                        <MyInput name="newPassword1" 	label={changePasswordMsg.newPassword} 		type="password" 	required/>
						<MyInput name="newPassword2" 	label={changePasswordMsg.retypeNewPassword} type="password" 	required/>
                    </MyForm>}
					{isError && <Alerts errors={ resetPasswordErrors } />}
					{isLoading && <Alerts errors={"changingPassword"} color="success"/>}
					{infoMsg && <Alerts errors={infoMsg} color="success"/>}
				</ModalBody>
				<ModalFooter>
					<Button type="submit" form="formChangePassword" color="success"><FormattedMessage {...changePasswordMsg.changePassword} /></Button>
                    <Button color="secondary" onClick={cancel}><FormattedMessage {...changePasswordMsg.cancel} /></Button>
				</ModalFooter>
			</Modal>
		</CSSTransition>
	);
}

ModalPasswordReset.propTypes = {
    toggle: PropTypes.func.isRequired,
	visible: PropTypes.bool.isRequired,
	className: PropTypes.string,
	data: PropTypes.object
};

export default ModalPasswordReset;