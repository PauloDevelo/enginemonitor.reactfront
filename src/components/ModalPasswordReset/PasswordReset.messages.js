import {defineMessages} from "react-intl";
import {type Messages} from "./MessageDescriptor"

const messages: Messages = {
    email: {
        id: "email",
        defaultMessage: "Email",
        description: "Label"
    },
    newPassword: {
        id: "newPassword",
        defaultMessage: "New Password",
        description: "Label"
    },
    retypeNewPassword: {
        id: "retypeNewPassword",
        defaultMessage: "Retype your new Password",
        description: "Label"
    },
    changePassword: {
        id: "changePassword",
        defaultMessage: "Change Password",
        description: "Label"
    },
    modalResetPasswordTitle: {
        id: "modalResetPasswordTitle",
        defaultMessage: "Change your password",
        description: "Modal title"
    },
    cancel: {
		id: "cancel",
        defaultMessage: "Cancel",
		description: "Button cancel"
    },
    
};

const definedMessages: typeof messages = defineMessages(messages);

export default definedMessages;