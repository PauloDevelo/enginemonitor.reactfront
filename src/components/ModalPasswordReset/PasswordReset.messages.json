{
  "cancel": {
    "defaultMessage": "Cancel",
    "description": "Button cancel",
    "id": "cancel"
  },
  "changePassword": {
    "defaultMessage": "Change Password",
    "description": "Label",
    "id": "changePassword"
  },
  "close": {
    "defaultMessage": "Close",
    "description": "Button Close",
    "id": "close"
  },
  "email": {
    "defaultMessage": "Email",
    "description": "Label",
    "id": "email"
  },
  "modalResetPasswordTitle": {
    "defaultMessage": "Change your password",
    "description": "Modal title",
    "id": "modalResetPasswordTitle"
  },
  "newPassword": {
    "defaultMessage": "New Password",
    "description": "Label",
    "id": "newPassword"
  },
  "retypeNewPassword": {
    "defaultMessage": "Retype your new Password",
    "description": "Label",
    "id": "retypeNewPassword"
  }
}
