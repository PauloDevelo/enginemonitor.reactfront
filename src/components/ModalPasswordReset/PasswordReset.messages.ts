import {defineMessages, Messages} from "react-intl";

const messages: Messages = {
    email: {
        id: "email",
        defaultMessage: "Email",
        description: "Label"
    },
    newPassword: {
        id: "newPassword",
        defaultMessage: "New Password",
        description: "Label"
    },
    retypeNewPassword: {
        id: "retypeNewPassword",
        defaultMessage: "Retype your new Password",
        description: "Label"
    },
    changePassword: {
        id: "changePassword",
        defaultMessage: "Change Password",
        description: "Label"
    },
    modalResetPasswordTitle: {
        id: "modalResetPasswordTitle",
        defaultMessage: "Change your password",
        description: "Modal title"
    },
    cancel: {
		id: "cancel",
        defaultMessage: "Cancel",
		description: "Button cancel"
    },
    close: {
        id: "close",
        defaultMessage: "Close",
		description: "Button Close"
    }
};

const definedMessages: Messages = defineMessages(messages);

export default definedMessages;