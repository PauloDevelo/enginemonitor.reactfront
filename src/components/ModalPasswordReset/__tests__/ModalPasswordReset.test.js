import React from 'react';
import { mount } from 'enzyme';

// eslint-disable-next-line no-unused-vars
import localforage from 'localforage';

import ignoredMessages from '../../../testHelpers/MockConsole';

import userProxy from '../../../services/UserProxy';

import ModalPasswordReset from '../ModalPasswordReset';
import updateWrapper from '../../../testHelpers/EnzymeHelper';
import HttpError from '../../../http/HttpError';

jest.mock('../../../services/UserProxy');
jest.mock('localforage');

describe('ModalPasswordReset', () => {
  beforeAll(() => {
    ignoredMessages.length = 0;
    ignoredMessages.push('[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry. Using default message as fallback.');
    ignoredMessages.push('a test was not wrapped in act');
  });

  afterEach(() => {
    userProxy.resetPassword.mockReset();
  });

  const data = { email: 'paulodevelo@lovestreet.com', newPassword1: '', newPassword2: '' };

  it('should render the ModalPasswordReset', async () => {
    // Arrange
    const toggle = jest.fn();

    // Act
    const modalPasswordReset = mount(<ModalPasswordReset visible toggle={toggle} data={data} />);
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(modalPasswordReset).toMatchSnapshot();
    expect(modalPasswordReset.find('ModalFooter').find('Button').length).toBe(2);

    const alerts = modalPasswordReset.find('Alerts');
    expect(alerts.length).toBe(0);
  });

  it('should close the modal if the user click Cancel', async () => {
    // Arrange
    const toggle = jest.fn();

    const modalPasswordReset = mount(<ModalPasswordReset visible toggle={toggle} data={data} />);
    await updateWrapper(modalPasswordReset);

    const cancelButton = modalPasswordReset.find('ModalFooter').find('Button').at(1);

    // Act
    cancelButton.simulate('click');
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(modalPasswordReset.find('ModalFooter').find('Button').length).toBe(2);

    const alerts = modalPasswordReset.find('Alerts');
    expect(alerts.length).toBe(0);

    expect(toggle).toBeCalledTimes(1);
  });

  it('should call resetPassword with all the data input when the user click on reset Password, then the modal should close if the user click the Close button', async () => {
    // Arrange
    const toggle = jest.fn();
    jest.spyOn(userProxy, 'resetPassword').mockImplementation(() => Promise.resolve());

    const modalPasswordReset = mount(<ModalPasswordReset visible toggle={toggle} data={data} />);
    await updateWrapper(modalPasswordReset);

    const myForm = modalPasswordReset.find('Memo(MyForm)');
    const inputs = myForm.find('input');
    inputs.at(0).simulate('change', { target: { value: data.email } });
    inputs.at(1).simulate('change', { target: { value: 'password1' } });
    inputs.at(2).simulate('change', { target: { value: 'password1' } });
    await updateWrapper(modalPasswordReset);

    // Act
    modalPasswordReset.find('Memo(MyForm)').simulate('submit');
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(modalPasswordReset.find('ModalFooter').find('Button').length).toBe(1);

    const alerts = modalPasswordReset.find('Alerts');
    expect(alerts.length).toBe(1);
    expect(alerts.props()).toEqual({ error: 'confirmPasswordChange', color: 'success' });
    expect(toggle).toBeCalledTimes(0);

    expect(userProxy.resetPassword).toBeCalledTimes(1);
    expect(userProxy.resetPassword.mock.calls[0][0]).toEqual(data.email);
    expect(userProxy.resetPassword.mock.calls[0][1]).toEqual('password1');

    // Act
    modalPasswordReset.find('Memo(MyForm)').simulate('submit');
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(toggle).toBeCalledTimes(1);
  });

  it('should display an error if the 2 passwords are not the same', async () => {
    // Arrange
    const toggle = jest.fn();
    jest.spyOn(userProxy, 'resetPassword').mockImplementation(() => Promise.resolve());

    const modalPasswordReset = mount(<ModalPasswordReset visible toggle={toggle} data={data} />);
    await updateWrapper(modalPasswordReset);

    const myForm = modalPasswordReset.find('Memo(MyForm)');
    const inputs = myForm.find('input');
    inputs.at(0).simulate('change', { target: { value: data.email } });
    inputs.at(1).simulate('change', { target: { value: 'password1' } });
    inputs.at(2).simulate('change', { target: { value: 'password2' } });
    await updateWrapper(modalPasswordReset);

    // Act
    modalPasswordReset.find('Memo(MyForm)').simulate('submit');
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(modalPasswordReset.find('ModalFooter').find('Button').length).toBe(2);

    const alerts = modalPasswordReset.find('Alerts');
    expect(alerts.length).toBe(1);
    expect(alerts.props()).toEqual({ errors: { password: 'passwordsHaveToBeIdentical' } });
    expect(toggle).toBeCalledTimes(0);

    expect(userProxy.resetPassword).toBeCalledTimes(0);
  });

  it('should display an error message if an error occurs during the password reset', async () => {
    // Arrange
    const toggle = jest.fn();
    jest.spyOn(userProxy, 'resetPassword').mockImplementation(() => Promise.reject(new HttpError({ error: 'network error' })));

    const modalPasswordReset = mount(<ModalPasswordReset visible toggle={toggle} data={data} />);
    await updateWrapper(modalPasswordReset);

    const myForm = modalPasswordReset.find('Memo(MyForm)');
    const inputs = myForm.find('input');
    inputs.at(0).simulate('change', { target: { value: data.email } });
    inputs.at(1).simulate('change', { target: { value: 'password1' } });
    inputs.at(2).simulate('change', { target: { value: 'password1' } });
    await updateWrapper(modalPasswordReset);

    // Act
    modalPasswordReset.find('Memo(MyForm)').simulate('submit');
    await updateWrapper(modalPasswordReset);

    // Assert
    expect(modalPasswordReset.find('ModalFooter').find('Button').length).toBe(2);

    const alerts = modalPasswordReset.find('Alerts');
    expect(alerts.length).toBe(1);
    expect(alerts.props()).toEqual({ errors: { error: 'network error' } });
    expect(toggle).toBeCalledTimes(0);

    expect(userProxy.resetPassword).toBeCalledTimes(1);
    expect(userProxy.resetPassword.mock.calls[0][0]).toEqual(data.email);
    expect(userProxy.resetPassword.mock.calls[0][1]).toEqual('password1');
  });
});
