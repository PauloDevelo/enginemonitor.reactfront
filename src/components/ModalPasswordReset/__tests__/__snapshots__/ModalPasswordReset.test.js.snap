// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPasswordReset should render the ModalPasswordReset 1`] = `
<ModalPasswordReset
  data={
    Object {
      "email": "paulodevelo@lovestreet.com",
      "newPassword1": "",
      "newPassword2": "",
    }
  }
  toggle={[MockFunction]}
  visible={true}
>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={true}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <Portal
      node={
        <div
          style="position: relative; z-index: 1050; display: block;"
          tabindex="-1"
        >
          <div
            class=""
          >
            <div
              class="modal fade"
              role="dialog"
              style="display: block;"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-unlock-alt fa-w-14 "
                        data-icon="unlock-alt"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 448 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zM264 408c0 22.1-17.9 40-40 40s-40-17.9-40-40v-48c0-22.1 17.9-40 40-40s40 17.9 40 40v48z"
                          fill="currentColor"
                        />
                      </svg>
                       
                      Change your password
                    </h5>
                    <button
                      aria-label="Close"
                      class="close"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <form
                      class=""
                      id="formChangePassword"
                      novalidate=""
                    >
                      <div
                        class="form-group form-group"
                      >
                        <label
                          class=""
                          for="email"
                        >
                          Email
                        </label>
                        <input
                          class="form-control form-control"
                          name="email"
                          readonly=""
                          required=""
                          type="email"
                          value="paulodevelo@lovestreet.com"
                        />
                      </div>
                      <div
                        class="form-group form-group"
                      >
                        <label
                          class=""
                          for="newPassword1"
                        >
                          New Password
                        </label>
                        <input
                          class="form-control form-control"
                          name="newPassword1"
                          required=""
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="form-group form-group"
                      >
                        <label
                          class=""
                          for="newPassword2"
                        >
                          Retype your new Password
                        </label>
                        <input
                          class="form-control form-control"
                          name="newPassword2"
                          required=""
                          type="password"
                          value=""
                        />
                      </div>
                    </form>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      class="btn btn-success"
                      form="formChangePassword"
                      type="submit"
                    >
                      Change Password
                    </button>
                    <button
                      class="btn btn-secondary"
                      type="button"
                    >
                      Cancel
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="modal-backdrop fade"
            />
          </div>
        </div>
      }
    >
      <Portal
        containerInfo={
          <div
            style="position: relative; z-index: 1050; display: block;"
            tabindex="-1"
          >
            <div
              class=""
            >
              <div
                class="modal fade"
                role="dialog"
                style="display: block;"
                tabindex="-1"
              >
                <div
                  class="modal-dialog"
                  role="document"
                >
                  <div
                    class="modal-content"
                  >
                    <div
                      class="modal-header"
                    >
                      <h5
                        class="modal-title"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-unlock-alt fa-w-14 "
                          data-icon="unlock-alt"
                          data-prefix="fas"
                          focusable="false"
                          role="img"
                          viewBox="0 0 448 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zM264 408c0 22.1-17.9 40-40 40s-40-17.9-40-40v-48c0-22.1 17.9-40 40-40s40 17.9 40 40v48z"
                            fill="currentColor"
                          />
                        </svg>
                         
                        Change your password
                      </h5>
                      <button
                        aria-label="Close"
                        class="close"
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          ×
                        </span>
                      </button>
                    </div>
                    <div
                      class="modal-body"
                    >
                      <form
                        class=""
                        id="formChangePassword"
                        novalidate=""
                      >
                        <div
                          class="form-group form-group"
                        >
                          <label
                            class=""
                            for="email"
                          >
                            Email
                          </label>
                          <input
                            class="form-control form-control"
                            name="email"
                            readonly=""
                            required=""
                            type="email"
                            value="paulodevelo@lovestreet.com"
                          />
                        </div>
                        <div
                          class="form-group form-group"
                        >
                          <label
                            class=""
                            for="newPassword1"
                          >
                            New Password
                          </label>
                          <input
                            class="form-control form-control"
                            name="newPassword1"
                            required=""
                            type="password"
                            value=""
                          />
                        </div>
                        <div
                          class="form-group form-group"
                        >
                          <label
                            class=""
                            for="newPassword2"
                          >
                            Retype your new Password
                          </label>
                          <input
                            class="form-control form-control"
                            name="newPassword2"
                            required=""
                            type="password"
                            value=""
                          />
                        </div>
                      </form>
                    </div>
                    <div
                      class="modal-footer"
                    >
                      <button
                        class="btn btn-success"
                        form="formChangePassword"
                        type="submit"
                      >
                        Change Password
                      </button>
                      <button
                        class="btn btn-secondary"
                        type="button"
                      >
                        Cancel
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="modal-backdrop fade"
              />
            </div>
          </div>
        }
      >
        <div
          className=""
        >
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onClick={[Function]}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            role="dialog"
            style={
              Object {
                "display": "block",
              }
            }
            tabIndex="-1"
            tag="div"
            timeout={300}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={300}
              unmountOnExit={false}
            >
              <div
                className="modal fade"
                onClick={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                role="dialog"
                style={
                  Object {
                    "display": "block",
                  }
                }
                tabIndex="-1"
              >
                <div
                  className="modal-dialog"
                  role="document"
                >
                  <div
                    className="modal-content"
                  >
                    <ModalHeader
                      charCode={215}
                      closeAriaLabel="Close"
                      tag="h5"
                      toggle={[Function]}
                      wrapTag="div"
                    >
                      <div
                        className="modal-header"
                      >
                        <h5
                          className="modal-title"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  448,
                                  512,
                                  Array [],
                                  "f13e",
                                  "M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zM264 408c0 22.1-17.9 40-40 40s-40-17.9-40-40v-48c0-22.1 17.9-40 40-40s40 17.9 40 40v48z",
                                ],
                                "iconName": "unlock-alt",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-unlock-alt fa-w-14 "
                              data-icon="unlock-alt"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 448 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48zM264 408c0 22.1-17.9 40-40 40s-40-17.9-40-40v-48c0-22.1 17.9-40 40-40s40 17.9 40 40v48z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                           
                          <FormattedMessage
                            defaultMessage="Change your password"
                            description="Modal title"
                            id="modalResetPasswordTitle"
                            values={Object {}}
                          >
                            Change your password
                          </FormattedMessage>
                        </h5>
                        <button
                          aria-label="Close"
                          className="close"
                          onClick={[Function]}
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                        </button>
                      </div>
                    </ModalHeader>
                    <ModalBody
                      tag="div"
                    >
                      <div
                        className="modal-body"
                      >
                        <Memo(MyForm)
                          id="formChangePassword"
                          initialData={
                            Object {
                              "email": "paulodevelo@lovestreet.com",
                              "newPassword1": "",
                              "newPassword2": "",
                            }
                          }
                          submit={[Function]}
                        >
                          <Form
                            id="formChangePassword"
                            innerRef={
                              Object {
                                "current": <form
                                  class=""
                                  id="formChangePassword"
                                  novalidate=""
                                >
                                  <div
                                    class="form-group form-group"
                                  >
                                    <label
                                      class=""
                                      for="email"
                                    >
                                      Email
                                    </label>
                                    <input
                                      class="form-control form-control"
                                      name="email"
                                      readonly=""
                                      required=""
                                      type="email"
                                      value="paulodevelo@lovestreet.com"
                                    />
                                  </div>
                                  <div
                                    class="form-group form-group"
                                  >
                                    <label
                                      class=""
                                      for="newPassword1"
                                    >
                                      New Password
                                    </label>
                                    <input
                                      class="form-control form-control"
                                      name="newPassword1"
                                      required=""
                                      type="password"
                                      value=""
                                    />
                                  </div>
                                  <div
                                    class="form-group form-group"
                                  >
                                    <label
                                      class=""
                                      for="newPassword2"
                                    >
                                      Retype your new Password
                                    </label>
                                    <input
                                      class="form-control form-control"
                                      name="newPassword2"
                                      required=""
                                      type="password"
                                      value=""
                                    />
                                  </div>
                                </form>,
                              }
                            }
                            noValidate={true}
                            onSubmit={[Function]}
                            tag="form"
                          >
                            <form
                              className=""
                              id="formChangePassword"
                              noValidate={true}
                              onSubmit={[Function]}
                            >
                              <Memo()
                                handleChange={[Function]}
                                key=".0"
                                label={
                                  Object {
                                    "defaultMessage": "Email",
                                    "description": "Label",
                                    "id": "email",
                                  }
                                }
                                name="email"
                                readOnly={true}
                                required={true}
                                type="email"
                                validationTrigger={0}
                                value="paulodevelo@lovestreet.com"
                              >
                                <FormGroup
                                  className="form-group"
                                  tag="div"
                                >
                                  <div
                                    className="form-group form-group"
                                  >
                                    <Label
                                      for="email"
                                      tag="label"
                                      widths={
                                        Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
                                      }
                                    >
                                      <label
                                        className=""
                                        htmlFor="email"
                                      >
                                        <FormattedMessage
                                          defaultMessage="Email"
                                          description="Label"
                                          id="email"
                                          values={Object {}}
                                        >
                                          Email
                                        </FormattedMessage>
                                      </label>
                                    </Label>
                                    <Input
                                      className="form-control"
                                      innerRef={
                                        Object {
                                          "current": <input
                                            class="form-control form-control"
                                            name="email"
                                            readonly=""
                                            required=""
                                            type="email"
                                            value="paulodevelo@lovestreet.com"
                                          />,
                                        }
                                      }
                                      invalid={false}
                                      name="email"
                                      onChange={[Function]}
                                      readOnly={true}
                                      required={true}
                                      type="email"
                                      value="paulodevelo@lovestreet.com"
                                    >
                                      <input
                                        className="form-control form-control"
                                        name="email"
                                        onChange={[Function]}
                                        readOnly={true}
                                        required={true}
                                        type="email"
                                        value="paulodevelo@lovestreet.com"
                                      />
                                    </Input>
                                  </div>
                                </FormGroup>
                              </Memo()>
                              <Memo()
                                handleChange={[Function]}
                                key=".1"
                                label={
                                  Object {
                                    "defaultMessage": "New Password",
                                    "description": "Label",
                                    "id": "newPassword",
                                  }
                                }
                                name="newPassword1"
                                required={true}
                                type="password"
                                validationTrigger={0}
                                value=""
                              >
                                <FormGroup
                                  className="form-group"
                                  tag="div"
                                >
                                  <div
                                    className="form-group form-group"
                                  >
                                    <Label
                                      for="newPassword1"
                                      tag="label"
                                      widths={
                                        Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
                                      }
                                    >
                                      <label
                                        className=""
                                        htmlFor="newPassword1"
                                      >
                                        <FormattedMessage
                                          defaultMessage="New Password"
                                          description="Label"
                                          id="newPassword"
                                          values={Object {}}
                                        >
                                          New Password
                                        </FormattedMessage>
                                      </label>
                                    </Label>
                                    <Input
                                      className="form-control"
                                      innerRef={
                                        Object {
                                          "current": <input
                                            class="form-control form-control"
                                            name="newPassword1"
                                            required=""
                                            type="password"
                                            value=""
                                          />,
                                        }
                                      }
                                      invalid={false}
                                      name="newPassword1"
                                      onChange={[Function]}
                                      required={true}
                                      type="password"
                                      value=""
                                    >
                                      <input
                                        className="form-control form-control"
                                        name="newPassword1"
                                        onChange={[Function]}
                                        required={true}
                                        type="password"
                                        value=""
                                      />
                                    </Input>
                                  </div>
                                </FormGroup>
                              </Memo()>
                              <Memo()
                                handleChange={[Function]}
                                key=".2"
                                label={
                                  Object {
                                    "defaultMessage": "Retype your new Password",
                                    "description": "Label",
                                    "id": "retypeNewPassword",
                                  }
                                }
                                name="newPassword2"
                                required={true}
                                type="password"
                                validationTrigger={0}
                                value=""
                              >
                                <FormGroup
                                  className="form-group"
                                  tag="div"
                                >
                                  <div
                                    className="form-group form-group"
                                  >
                                    <Label
                                      for="newPassword2"
                                      tag="label"
                                      widths={
                                        Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
                                      }
                                    >
                                      <label
                                        className=""
                                        htmlFor="newPassword2"
                                      >
                                        <FormattedMessage
                                          defaultMessage="Retype your new Password"
                                          description="Label"
                                          id="retypeNewPassword"
                                          values={Object {}}
                                        >
                                          Retype your new Password
                                        </FormattedMessage>
                                      </label>
                                    </Label>
                                    <Input
                                      className="form-control"
                                      innerRef={
                                        Object {
                                          "current": <input
                                            class="form-control form-control"
                                            name="newPassword2"
                                            required=""
                                            type="password"
                                            value=""
                                          />,
                                        }
                                      }
                                      invalid={false}
                                      name="newPassword2"
                                      onChange={[Function]}
                                      required={true}
                                      type="password"
                                      value=""
                                    >
                                      <input
                                        className="form-control form-control"
                                        name="newPassword2"
                                        onChange={[Function]}
                                        required={true}
                                        type="password"
                                        value=""
                                      />
                                    </Input>
                                  </div>
                                </FormGroup>
                              </Memo()>
                            </form>
                          </Form>
                        </Memo(MyForm)>
                      </div>
                    </ModalBody>
                    <ModalFooter
                      tag="div"
                    >
                      <div
                        className="modal-footer"
                      >
                        <ActionButton
                          color="success"
                          form="formChangePassword"
                          isActing={false}
                          message={
                            Object {
                              "defaultMessage": "Change Password",
                              "description": "Label",
                              "id": "changePassword",
                            }
                          }
                          type="submit"
                        >
                          <Button
                            color="success"
                            form="formChangePassword"
                            tag="button"
                            type="submit"
                          >
                            <button
                              aria-label={null}
                              className="btn btn-success"
                              form="formChangePassword"
                              onClick={[Function]}
                              type="submit"
                            >
                              <FormattedMessage
                                defaultMessage="Change Password"
                                description="Label"
                                id="changePassword"
                                values={Object {}}
                              >
                                Change Password
                              </FormattedMessage>
                            </button>
                          </Button>
                        </ActionButton>
                        <Button
                          color="secondary"
                          onClick={[Function]}
                          tag="button"
                        >
                          <button
                            aria-label={null}
                            className="btn btn-secondary"
                            onClick={[Function]}
                            type="button"
                          >
                            <FormattedMessage
                              defaultMessage="Cancel"
                              description="Button cancel"
                              id="cancel"
                              values={Object {}}
                            >
                              Cancel
                            </FormattedMessage>
                          </button>
                        </Button>
                      </div>
                    </ModalFooter>
                  </div>
                </div>
              </div>
            </Transition>
          </Fade>
          <Fade
            appear={true}
            baseClass="fade"
            baseClassActive="show"
            className="modal-backdrop"
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={150}
            unmountOnExit={false}
          >
            <Transition
              appear={true}
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={150}
              unmountOnExit={false}
            >
              <div
                className="modal-backdrop fade"
              />
            </Transition>
          </Fade>
        </div>
      </Portal>
    </Portal>
  </Modal>
</ModalPasswordReset>
`;
